### PR TITLE
Refactor Header tests

### PR DESCRIPTION
**Pull request by evadb**
This PR adds a better test suite to the Header component. I've also changed the 'is' prop to be required by the Header component. This is because this prop sets the tag the Header will render as (h1, h2, h3 etc) and the Header component should never be used without it.

### CODE REVIEW

File: frontend/src/components/header/__tests__/Header.test.js [hunk 1]
```diff
@@ -1,59 +1,115 @@
...
+  })
+
+  it("The 'is' prop sets correct header tag", () => {
+    let wrapper = mount(<Header is="h1" />).find('h1')
+    expect(wrapper.is('h1')).toBeTruthy()
+  })
+
+  it('H1 preset renders as h1', () => {
+    let wrapper = mount(<H1 />).find('h1')
+    expect(wrapper.is('h1')).toBeTruthy()
```
Comment (sastels) [inline frontend/src/components/header/__tests__/Header.test.js]: this is nice!

File: frontend/src/components/header/__tests__/Header.test.js [hunk 2]
```diff
@@ -1,59 +1,115 @@
...
+        foo
+      </Header>,
     )
+
+    const test = getByText('foo')
+    expect(test).toHaveStyleRule('font-weight', 'bold')
+  })
+
+  it("The 'is' prop sets correct header tag", () => {
+    let wrapper = mount(<Header is="h1" />).find('h1')
```
Comment (sastels) [inline frontend/src/components/header/__tests__/Header.test.js]: Do you need all these `find`s here and in subsequent tests? Isn't wrapper just going to be the entire thing anyway?

Comment (evadb) [reply frontend/src/components/header/__tests__/Header.test.js]: If `shallow` rendering yes. Unfortunately I can't seem to use Shallow rendering for these? :( So I used `mount` which wraps it in a bunch of divs. So I used the finds to pick up the header

File: frontend/src/components/header/__tests__/Header.test.js (hunk 1 above)
Comment (evadb) [reply frontend/src/components/header/__tests__/Header.test.js]: shamelessly stolen from @sleepycat

File: frontend/src/components/header/__tests__/Header.test.js (hunk 2 above)
Comment (sastels) [reply frontend/src/components/header/__tests__/Header.test.js]: wait a sec, why are you using `enzyme` here, can't something from `react-testing-library` do the job?

Comment (evadb) [reply frontend/src/components/header/__tests__/Header.test.js]: I spent a while reading docs and looking things up, but was unable to find a matcher to check for HTML tag type. Enzyme had one so I decided to just use that

**Pull request merged**